index.js

```diff
@@ -99,7 +99,11 @@ function lifecycle (pkg, stage, wd, opts) {
 
         // 'nobody' typically doesn't have permission to write to /tmp
         // even if it's never used, sh freaks out.
-        if (!opts.unsafePerm) env.TMPDIR = wd
+        if (!opts.unsafePerm) {
+          const tmpdir = path.join(wd, 'node_modules')
+          if (!fs.existsSync(tmpdir)) fs.mkdirSync(tmpdir)
```

Comment:
It is better to just run mkdirSync and skip the EEXIST error. Something like this
```ts
try {
  fs.mkdirSync(tmpDir)
} catch (err) {
  if (err.code !== 'EEXIST') throw err
}
```